tests/test_sync.py

```diff
@@ -257,8 +260,65 @@ def test_get_sync_work_state(self):
                 with self.subTest(part_size=part_size, object_size=event['source_obj_metadata']['size']):
                     self.assertEqual(sync.get_sync_work_state(event)['total_parts'], expected_total_parts)
 
+    def test_blob_key_format(self):
+        good_blob_key = 'blobs/015b0751073d17e358989649bc49f7e2bcf544073a98bf24d762fbe5c6468cb3.' \
+                        'd3c4a8f1aadcf6036c9c7ca5e80189fce3f12629.70910bba1cc60b415f34254f7e0d9673.93d22640'
+        bad_blob_key = 'blobs/5b0751073d17e358989649bc49f7e2bcf544073a98bf24d762fbe5c6468cb3.' \
+                       'd3c4a8f1aadcf6036c9c7ca5e80189fce3f12629.70910bba1cc60b415f34254f7e0d9673.93d2264'
+        missing_prefix = '015b0751073d17e358989649bc49f7e2bcf544073a98bf24d762fbe5c6468cb3.' \
+                         'd3c4a8f1aadcf6036c9c7ca5e80189fce3f12629.70910bba1cc60b415f34254f7e0d9673.93d22640'
+        gcp_parts = 'blobs/015b0751073d17e358989649bc49f7e2bcf544073a98bf24d762fbe5c6468cb3.' \
+                    'd3c4a8f1aadcf6036c9c7ca5e80189fce3f12629.70910bba1cc60b415f34254f7e0d9673.93d22640.partA'
```

Comment:
This looks fine, but it would be even better to produce random good and bad keys. That increases the chance we'll discover some weird bug in our regex.

Reply:
added random key generation